src/widgets/Autocomplete/index.tsx

```diff
@@ -0,0 +1,216 @@
+}
+
+// TODO: autofocus after unselecting an item
+
+const Autocomplete = <T extends any>({
+  children, debounce = 250, deps = [], disabled = false, getItems, initialQuery = '',
+  itemToString, loadingPrompt = 'Loading...', noResults = 'No results', renderItem,
+  required = false, value, onBlur, onChange, onFocus, ...rest
+}: Props<T>) => {
+  const [query, setQuery] = useState(initialQuery)
+  const [activeIndex, setActiveIndex] = useState(0)
+  const [suggestions, setSuggestions] = useState<Array<Suggestion<T>>>([])
+  const [isFocused, setIsFocused] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const debouncedQuery = useDebouncedValue(query, debounce)
+
+  useEffect(() => {
+    if (disabled) return
+
+    setIsLoading(true)
+    setActiveIndex(0)
+
+    debug.log('[Autocomplete] getItems')
+
+    getItems(debouncedQuery)
+      .then((items) => {
+        setSuggestions(items.map((el, index): Suggestion<T> => ({
+          active: index === 0, index, item: el, onSelect: () => handleSelect(el)
+        })))
+        setIsLoading(false)
+      })
+      .catch((error) => {
+        debug.error('[Autocomplete] getItems =>', error)
+      })
+  }, [disabled, debouncedQuery, ...deps])
+
+  useMemo(() => {
+    setSuggestions(suggestions => suggestions.map((suggestion, index) => ({
+      ...suggestion, active: index === activeIndex
+    })))
+  }, [activeIndex])
+
+  const handleBlur = useCallback(() => {
+    setActiveIndex(0)
+    setIsFocused(false)
+    if (onBlur) onBlur()
+  }, [onBlur])
+
+  const handleFocus = useCallback(() => {
+    setIsFocused(true)
+    if (onFocus) onFocus()
+  }, [onFocus])
+
+  const handleChange = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
+    setQuery(e.currentTarget.value)
+
+    if (required) {
+      e.currentTarget.setCustomValidity('Please select an option')
+    }
+  }, [required])
+
+  const handleKeyDown = useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.keyCode === KeyCode.DOWN) {
+      e.preventDefault()
+      setActiveIndex(index => Math.min(index + 1, suggestions.length - 1))
+    } else if (e.keyCode === KeyCode.UP) {
+      e.preventDefault()
+      setActiveIndex(index => Math.max(index - 1, 0))
+    } else if (e.keyCode === KeyCode.ENTER) {
+      e.preventDefault()
+      handleSelect(suggestions[activeIndex].item)
+    }
+  }, [suggestions])
+
+  const handleSelect = useCallback((item: Selection<T>) => {
+    if (disabled) return
+
+    setActiveIndex(0)
+    onChange(item)
+  }, [disabled, onChange])
+
+  if (value) {
+    if (renderItem) {
+      return renderItem(value, () => handleSelect(null))
+    }
+
+    return (
+      <Item
+        disabled={disabled}
+        justify='space-between'
+        onClick={() => handleSelect(null)}
+      >
+        <Text color='grey800' size={0.875}>
+          {itemToString(value)}
+        </Text>
+        <Text size={0.75}>
+          <Icon name='times' />
```

Comment:
Icon now accepts an `icon` prop instead of `name` + `prefix`
```suggestion
          <Icon icon={{ iconName: 'times', prefix: 'far' }} />
```